core/deployment/src/main/java/io/quarkus/deployment/configuration/BuildTimeConfigurationReader.java

```diff
@@ -220,6 +220,11 @@ private BuildTimeConfigurationReader(ClassLoader classLoader, final List<Class<?
                 assert phase == ConfigPhase.RUN_TIME;
                 runTimeRoots.add(definition);
             }
+
+            if (configRoot.getAnnotation(Deprecated.class) == null) {
+                log.warn(configRoot.getName()
+                        + " is considered legacy and should be moved to use an interface and @ConfigMapping");
```

Comment:
It would be nice to have an URL to explain why and how to do that:

```suggestion
                        + " is considered legacy and should be moved to use an interface and @ConfigMapping. See https://quarkus.io/guides/config-mappings for more info");
```

Reply:
Yeah, the message needs some tweaking, I will have a look soon.

It might be a bit too early to push it anyway.